Src/Support/Google.Apis.Auth/OAuth2/CredentialFactory.cs

```diff
@@ -125,27 +221,53 @@ public static T FromJson<T>(string json)
     /// </exception>
     internal static T FromJsonParameters<T>(JsonCredentialParameters credentialParameters)
     {
-        if (CreateCredential(credentialParameters, typeof(T)) is T credentialAsT)
+        credentialParameters.ThrowIfNull(nameof(credentialParameters));
+        if (CreateCredentialFromParameters(credentialParameters, typeof(T)) is T credentialAsT)
         {
             return credentialAsT;
         }
 
         throw new InvalidOperationException(
             $"Found incompatible credential types, '{credentialParameters.Type}' and '{typeof(T).FullName}, even though a check" 
           + " should have already taken place. We should never reach here, there's a bug in the code.");
+    }
 
-        static IGoogleCredential CreateCredential(JsonCredentialParameters credentialParameters, Type targetCredentialType) =>
-            credentialParameters.ThrowIfNull(nameof(credentialParameters)).Type switch
-            {
-                JsonCredentialParameters.AuthorizedUserCredentialType => CreateUserCredentialFromParameters(credentialParameters, targetCredentialType),
-                JsonCredentialParameters.ServiceAccountCredentialType => CreateServiceAccountCredentialFromParameters(credentialParameters, targetCredentialType),
-                JsonCredentialParameters.ExternalAccountCredentialType => CreateExternalCredentialFromParameters(credentialParameters, targetCredentialType),
-                JsonCredentialParameters.ImpersonatedServiceAccountCredentialType => CreateImpersonatedServiceAccountCredentialFromParameters(credentialParameters, targetCredentialType),
-                JsonCredentialParameters.ExternalAccountAuthorizedUserCredentialType => CreateExternalAccountAuthorizedUserCredentialFromParameters(credentialParameters, targetCredentialType),
-                _ => throw new InvalidOperationException($"Error creating credential from JSON or JSON parameters. Unrecognized credential type {credentialParameters.Type}."),
-            };
+    /// <summary>
+    /// Creates a credential of the specified type from JSON credential parameters.
+    /// </summary>
+    /// <param name="credentialParameters">The JSON credential parameters.</param>
+    /// <param name="credentialType">The expected type of the credential.</param>
```

Comment:
nit: Add here a note similar to the one in the public methods "... as defined in JsonCredentialParameters"